python/dgl/distributed/dist_graph.py

```diff
@@ -85,11 +80,10 @@ def _get_graph_from_shared_mem(graph_name):
     The client can access the graph structure and some metadata on nodes and edges directly
     through shared memory to reduce the overhead of data access.
     '''
-    gidx = from_shared_mem_graph_index(_get_graph_path(graph_name))
-    if gidx is None:
-        return gidx
-
-    g = DGLGraph(gidx)
+    g, ntypes, etypes = heterograph_index.create_heterograph_from_shared_memory(graph_name)
+    if g is None:
+        return None
+    g = DGLHeteroGraph(g, ntypes, etypes)
```

Comment:
I think you can do this by `dgl.hetero_from_shared_memory` at https://github.com/soodoshll/dgl/blob/a7dbfd1341d30125e9f740f514aad10e34325510/python/dgl/convert.py#L418

Reply:
it's different. i need to deal with the case that shared-memory graph index doesn't exist. also i don't want to import from convert.py.